### ISSUE TITLE
Refactor NotificationsPanel React component for improved Accessibility and Composability

### ISSUE THREAD

**Issue by szinta** [carbon-design-system/ibm-products#8057]
Refactor the React NotificationsPanel to improve accessibility and make the component more composable

Background:

> The fix proposed in #7612 for accessibility violations works well for the Web Component version. However, in the React version, the notification component includes multiple interactive elements (e.g., close button, link, and "Read more" button), which can lead to `Nested interactive elements` violations.

The refinement should consider the following points:

- Creating a standalone sub-component for individual notification items.
- Making NotificationsPanel composable by removing embedded logic for "Read more" and link. These should be handled by consumers of the component.
- Replace the "Read more" button with the truncatedText utility to handle text overflow
